### PR TITLE
ci: report micro-benchmarking diff

### PR DESCRIPTION
**Pull request by v1v**
### What

This will enable the micro-benchmarking diff report on a PR basis if they are triggered.

### Further actions

A similar implementation was done in https://github.com/elastic/apm-server/pull/8471 but it required a couple of changes more:
- https://github.com/elastic/apm-server/pull/8472
- https://github.com/elastic/apm-server/pull/8476

### Question

Do we want something similar for the APM Agent Go?

### CODE REVIEW

**Review by cachedout — APPROVED** [review]

**Comment by v1v** [comment]
Requires https://github.com/elastic/apm-agent-go/pull/1265 otherwise the MacOS stage will keep failing

**Review by axw — APPROVED** [review]

**Comment by v1v** [comment]
run benchmark tests

**Comment by v1v** [comment]
run benchmark tests

**Comment by v1v** [comment]
run benchmark tests

File: Jenkinsfile [hunk 1]
```diff
@@ -151,6 +151,10 @@ pipeline {
               dir("${BASE_DIR}"){
                 sh script: './scripts/jenkins/bench.sh', label: 'Benchmarking'
                 sendBenchmarks(file: 'build/bench.out', index: 'benchmark-go')
+                // generateGoBenchmarkDiff doesn't work well with nested folders.
+                dir('build') {
+                  generateGoBenchmarkDiff(file: 'bench.out', filter: 'exclude')
+                }
```
Comment (v1v) [inline Jenkinsfile:157]:
```suggestion
                // generateGoBenchmarkDiff doesn't work well with nested folders.
                dir('build') {
                  sendBenchmarks(file: 'bench.out', index: 'benchmark-go')
                  generateGoBenchmarkDiff(file: 'bench.out', filter: 'exclude')
                }
```

**Comment by v1v** [comment]
run benchmark tests

**Comment by marclop** [comment]
just checking, has this PR fallen through the cracks? 🙂

**Comment by v1v** [comment]
run benchmark tests

File: Jenkinsfile [hunk 2]
```diff
@@ -151,6 +151,7 @@ pipeline {
               dir("${BASE_DIR}"){
                 sh script: './scripts/jenkins/bench.sh', label: 'Benchmarking'
                 sendBenchmarks(file: 'build/bench.out', index: 'benchmark-go')
+                generateGoBenchmarkDiff(file: 'build/bench.out', filter: 'exclude')
```
Comment (v1v) [inline Jenkinsfile:154]: I'm not sure if `file` should be a filename or a relative path with nested folders...

**Comment by v1v** [comment]
I don't know what's the reason for the existing failures... I'd assume some kind of environmental issues?

**Pull request merged**